components/merkledb/src/views/metadata.rs

```diff
@@ -164,7 +164,7 @@ impl Default for IndexType {
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub struct IndexMetadata<V = Vec<u8>> {
     // A globally unique numeric index identifier. MerkleDB assigns a unique numeric ID for each
-    // fully-qualified index name.
+    // fully-qualified index name. Valid identifiers are non-zero.
```

Comment:
Not sure about this. This is not an invariant guaranteed by `IndexMetadata`, but rather an invariant of code that *use* `IndexMetadata`. That's a deep details and sort of verbalism though.

Reply:
I see, without enforcing that invariant it is not reliable. We could postpone this doc update till this made a strict invariant, so that we can rely on it.